clang/tools/scan-build-py/lib/libscanbuild/intercept.py

```diff
@@ -290,6 +302,29 @@ def abspath(cwd, name):
                 'command': encode(command),
             }
 
+# SYCLomatic_CUSTOMIZATION begin
+        # If linker entry contains source compilation files, generate output files for the source files, then generate linker entry
+        elif len(get_object_files(compilation.flags)) > 1 and compilation.files:
+            output_files = []
+            for source_idx in reversed(range(len(compilation.files))):
+                source = compilation.files[source_idx]
+                command = [compiler, "-c"] + filter_linker_entry(compilation.flags) + ['-o'] + [
+                    os.path.splitext(os.path.basename(source))[0]+'.o'] + [source]
+                output_files.append(os.path.splitext(
+                    os.path.basename(source))[0]+'.o')
```

Comment:
Pls make sure the code style consistent with current script code style, for example there should be one space before and after operator "+" in Line314.

Reply:
Fixed + operator to have consistent space.